adr/2025-10-23-pin-npm-dependencies.md

```diff
@@ -0,0 +1,56 @@
+---
+title: Pin All NPM Dependencies to Exact Versions
+date: 2025-10-23
+area: infrastructure
+tags: [repository, workflow, npm, dependencies, ci, security]
+---
+
+## Context
+
+NPM dependencies with range specifiers (e.g., `^1.2.3` or `~1.2.3`) allow automatic updates to newer compatible versions during installation. While convenient for library development, this creates several risks for application development:
+
+1. **Non-Deterministic Builds**: Different developers or CI runs may install different versions, leading to "works on my machine" issues
+2. **Security Risk Window**: Malicious packages can be introduced through automatic version updates without explicit review
+3. **Breaking Changes**: Even minor/patch updates can introduce bugs or breaking changes despite semantic versioning
+4. **Difficult Debugging**: Inconsistent dependency versions across environments make issues harder to reproduce and diagnose
+
+## Decision
+
+All NPM dependencies in Shopware 6 must be pinned to exact versions without range specifiers:
+- ❌ `"package": "^1.2.3"` or `"package": "~1.2.3"`
```

Comment:
see my comment above, there are more ways to not specify an exact (pinned) version